src/Notes.cc

```diff
@@ -354,6 +354,15 @@ NotePairs::appendNewOnly(const NotePairs *src)
     }
 }
 
+void
+NotePairs::replaceOrAddOrAppend(const NotePairs *src, const std::vector<std::string> &appendKeys)
+{
+    for (auto e: src->entries)
+        if (std::find(appendKeys.begin(), appendKeys.end(), const_cast<SBuf &>(e->name()).c_str()) == appendKeys.end())
```

Comment:
The cast here is trying to go from char* to a reference of a complex object - that is wrong. Ideally the vector and the e->name() would use the same type and avoid any conversion at all. But our old String class does not do early initialization properly so until the notes member is converted to SBuf (dont do that here) this has to be a little bit messy.

Once the vector definition is using SBuf you should be able to use StringToSBuf(e->name()) to convert safely. That may need sbuf/StringConvert.h included to build, but try without that header first.

Reply:
Done! Also sbuf/StringConvert.h is already included in Notes.cc.

Just wondering if StringToSBuf() needs free() somewhere? But then I have no idea about vectors. Do they get freed auto-magically?

Reply:
All these objects handle their own memory.

Sorry I steered you a bit wrong earlier. It looks like the StringToSBuf() appears not to be needed at all.

Reply:
Oh yes! Since I was working on 4.7 branch, where name and value of NotePairs are String variables. So it needed StringToSBuf().

But I didnt notice that in master branch, name and value of NotePairs are already migrated to SBuf.

Fixed now. Fingers crossed!